services/github/github-api-provider.js

```diff
@@ -154,14 +185,14 @@ class GithubApiProvider {
   // Act like request(), but tweak headers and query to avoid hitting a rate
   // limit. Inject `request` so we can pass in `cachingRequest` from
   // `request-handler.js`.
-  request(request, url, options = {}, callback) {
+  request({ request, url, options = {}, neededScopes = {}, callback }) {
```

Comment:
Perhaps somewhat a subjective/stylistic preference, but I get uneasy having to deal with positional based parameters once you have more than 2 or 3, so opted to switch the signature to an objected/named params which of course introduces diffs at respective call sites (which is the source of most of the diff in the test files in this commit as well as the update to the suggestion service)